tests/capabilities.test.js

```diff
@@ -0,0 +1,230 @@
+const { arrayOf, waitForResource } = require('./utils');
+const { initialize, listDatasets, clearDataset } = require('./initialize');
+const { delay } = require('@semapps/ldp');
+const CapabilitiesService = require('@activitypods/core/services/capabilities');
+
+/**
+ * @typedef {import('moleculer').ServiceBroker} Broker
+ */
+jest.setTimeout(30000);
+
+const NUM_USERS = 2;
+
+/** @type {Broker} */
+let broker;
+
+const signupUser = async (num) => {
+  const { webId, token } = await broker.call('auth.signup', {
+    username: `user${num}`,
+    email: `user${num}@test.com`,
+    password: 'test',
+    name: `User #${num}`,
+  });
+  return { webId, token };
+};
+
+/** @param {Broker} broker @param {number} num */
+const createUser = async (broker, num) => {
+  const { webId, token } = await signupUser(num);
+
+  const webIdDoc = await waitForResource(
+    2000,
+    ['url', 'apods:capabilities'],
+    8,
+    async () =>
+      await broker.call('ldp.resource.get', {
+        resourceUri: webId,
+        accept: MIME_TYPES.JSON,
+        webId: 'system',
+      })
+  );
+
+  // The .id check is only necessary as long as the @context is not updated to contain `apods:capabilities`
+  const capsUri =
+    typeof webIdDoc['apods:capabilities'] === 'string'
+      ? webIdDoc['apods:capabilities']
+      : webIdDoc['apods:capabilities'].id;
+
+  return { uri: webId, profileUri: webIdDoc['url'], capabilitiesUri: capsUri, token };
+};
+
+const getUserInviteCap = async (user) => {
+  // Get all existing caps
+  const inviteCaps = await broker.call('ldp.container.get', {
+    containerUri: user.capabilitiesUri,
+    accept: MIME_TYPES.JSON,
+    webId: 'system',
+  });
+  const caps = arrayOf(inviteCaps['ldp:contains']);
+  const inviteCap = caps.find((cap) => {
+    return (
+      cap.type === 'http://www.w3.org/ns/auth/acl#Authorization' &&
+      cap['http://www.w3.org/ns/auth/acl#Mode'] === 'http://www.w3.org/ns/auth/acl#Read' &&
+      cap['http://www.w3.org/ns/auth/acl#AccessTo'] === user.profileUri
+    );
+  });
+  return inviteCap;
+};
+
+describe('capabilities', () => {
+  /**
+   * @typedef UserField
+   * @property {string} uri
+   * @property {string} profileUri
+   * @property {string} capabilitiesUri
+   * @property {string} token
+   */
+  /** @type {UserField[]} */
+  let users = [];
+
+  beforeAll(async () => {
+    const datasets = await listDatasets();
+    for (let dataset of datasets) {
+      await clearDataset(dataset);
+    }
+
+    broker = await initialize(3000, 'settings');
+
+    broker.loadService(path.resolve(__dirname, './services/profiles.app.js'));
+
+    await broker.start();
+
+    const newUserPromises = [];
+    for (let i = 1; i <= NUM_USERS; i++) {
+      newUserPromises.push(createUser(broker, i));
+    }
+
+    users = await Promise.all(newUserPromises);
+
+    // We still need to wait here, because not all `auth.registered` events
+    //  might have gone through. Is there a cleaner way?
+    await delay(5000);
```

Comment:
The `wait-for-expect` library, used on other SemApps tests, allow to wait for side-effects in most cases, reducing the wait time to the minimum. There are only some cases when this solution is not enough, and a delay must be added.